git_guidelines.md

```diff
@@ -111,104 +113,26 @@ As can be seen, the two pull requests are still distinguishable but the history
-```
-* |
-|\ \
-| * |
-* | |
-|\ \ \
-| |/ /
-|/| |
-| * |
-| * |
-| * |
-| * |
-| * |
-|/ /
-* |
-|\ \
-| |/
-|/|
-| *
-| *
-| *
-|/
-*
-```
+- GitHub's `Update Branch` button SHOULD not be used, `git pull --rebase` SHOULD
+  be used instead.
 
-Instead of using the `Update Branch` button use `git pull --rebase`. For the
-following example, we'll assume that we are working in a branch called
-`feature_xyz` that should be merged into the branch `master`. Furthermore the
-remote `origin` is a fork of exiv2 and the remote `upstream` is the "official"
-exiv2 repository.
 
-Before we start working, the `master` branch looks like this:
-```
-$ git log master --oneline --graph
-*   efee9a2b (master) Merge pull request #something
-|\
-| * ead7f309 A commit on master
-|/
-*   55001c8d Merge pull request #something else
-```
+## Merging pull requests
 
-We create a new branch `feature_xyz` based on `master`, create two new commits
-`My commit 1` and `My commit 2` and submit a pull request into master. The log
-of the branch `feature_xyz` now looks like this:
-```
-$ git log feature_xyz --oneline --graph
-* 893fffa5 (HEAD -> feature_xyz) My commit 2
-* a2a22fb9 My commit 1
-*   efee9a2b (master) Merge pull request #something
-|\
-| * ead7f309 A commit on master
-|/
-*   55001c8d Merge pull request #something else
-```
+- Most pull requests SHOULD be merged by creating a merge commit (the default on
+  GitHub).
 
-If now new commits are pushed to `master`, resulting in this log:
-```
-$ git log master --oneline --graph
-* 0d636cc9 (HEAD -> master) Hotfix for issue #something completely different
-*   efee9a2b Merge pull request #something
-|\
-| * ead7f309 A commit on master
-|/
-*   55001c8d Merge pull request #something else
-```
-then the branch `feature_xyz` is out of date with `master`, because it lacks the
-commit `0d636cc9`. We could now merge both branches (via the cli or GitHub's
-`Update Branch` button), but that will result in a messy history. Thus **don't**
-do it! If you do it, you'll have to remove the merge commits manually.
-
-Instead run: `git pull --rebase upstream master` in the `feature_xyz`
-branch. Git will pull the new commit `0d636cc9` from master into your branch
-`feature_xyz` and apply the two commits `My commit 1` and `My commit 2` on top
-of it:
-```
-$ git log feature_xyz --oneline --graph
-* 22a7a8c2 (HEAD -> feature_xyz) My commit 2
-* efe2ccdc My commit 1
-* 0d636cc9 (master) Hotfix for issue #something completely different
-*   efee9a2b Merge pull request #something
-|\
-| * ead7f309 A commit on master
-|/
-*   55001c8d Merge pull request #something else
-```
-Please note, that the hash of `My commit 1` and `My commit 2` changed! That
-happened because their parent changed. Therefore you have to force push your
-changes via `git push --force` next time you push your changes upstream.
+- Small pull requests (= only one commit) CAN be rebased.
 
+- Pull requests with GPG signed commits MUST NOT be rebased, as that will drop
```

Comment:
What should we do in this case then?

Reply:
By that wording: you can do nothing and that PR must be abbandoned :wink: 

I've changed it so that only the author should rebase, so that their signature is kept.

Reply:
hehe now it makes more sense 😛